### PR TITLE
Move PinVersions into RepoTasks

### PR DESCRIPTION
**Pull request by natemcmaster**
Removes this repo's dependency on .NET Core 1.1.

cref https://github.com/aspnet/BuildTools/issues/347

### CODE REVIEW

File: build/tasks/PinVersions.cs [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+                ? new DependencyGraphSpecProvider(GraphSpecsRoot)
+                : DependencyGraphSpecProvider.Default;
+
+            var sources = new List<string>();
+            sources.AddRange(PackageSources.Select(i => i.ItemSpec));
+
+            foreach (var item in AdditionalPackageSources)
+            {
+                var path = item.ItemSpec;
+                if (path.StartsWith("http", StringComparison.OrdinalIgnoreCase))
```
Comment (pranavkm) [inline build/tasks/PinVersions.cs]: This feels weird. It'd be cleaner to construct `PackageSources` in the target with the right set of sources instead.

File: build/tasks/PinVersions.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+        public override bool Execute()
+        {
+            if (PackageSources?.Length == 0)
+            {
+                Log.LogError($"Missing PackageSources. At least one item source must be specified.");
+                return false;
+            }
+
+            if (string.IsNullOrEmpty(BuildRepositoryRoot))
```
Comment (pranavkm) [inline build/tasks/PinVersions.cs]: Does `Required` not cover empty strings?

File: build/tasks/RepoTasks.tasks [hunk 3]
```diff
@@ -4,4 +4,5 @@
   </PropertyGroup>
 
   <UsingTask TaskName="RepoTasks.CalculateBuildGraph" AssemblyFile="$(_RepoTaskAssembly)" />
+  <UsingTask TaskName="RepoTasks.PinVersions" AssemblyFile="$(_RepoTaskAssembly)" />
```
Comment (pranavkm) [inline build/tasks/RepoTasks.tasks]: New line at EOF as long as you're editing this file.

File: build/tasks/PinVersions.cs (hunk 1 above)
Comment (natemcmaster) [inline build/tasks/PinVersions.cs]: updated

File: build/tasks/PinVersions.cs (hunk 2 above)
Comment (natemcmaster) [inline build/tasks/PinVersions.cs]: It does. Updated to remove the check

**Pull request merged**